packages/runtime/container-runtime/src/opLifecycle/opDecompressor.ts

```diff
@@ -122,3 +119,34 @@ const newMessage = (
 	compression: undefined,
 	metadata: { ...originalMessage.metadata },
 });
+
+const isCompressed = (message: ISequencedDocumentMessage): boolean => {
+	if (message.compression === CompressionAlgorithms.lz4) {
+		return true;
+	}
+
+	// This condition holds true for compressed messages, regardless of metadata.
+	// Back-compat self healing mechanism for ADO:3538, as loaders from
+	// version client_v2.0.0-internal.1.2.0 to client_v2.0.0-internal.2.2.0 do not
+	// support adding the proper compression metadata to compressed messages submitted
+	// by the runtime. Should be removed after the loader reaches sufficient saturation
+	// for a version greater or equal than client_v2.0.0-internal.2.2.0.
+	if (typeof message.contents === "object" && message.contents?.packedContents !== undefined) {
```

Comment:
I'd suggest to have everything either as IF condition or everything as return condition (i.e. do not break check into two parts).